esmvalcore/_recipe.py

```diff
@@ -1130,6 +1122,16 @@ def _split_derive_profile(profile):
     return before, after
 
 
+def _check_differing_timeranges(timeranges, required_vars):
+    """Log error if required variables have differing timeranges."""
+    if len(timeranges) > 1:
+        logger.error(
+            "Differing timeranges with values %s "
+            "found for required variables %s. "
+            "Set `timerange` to a common value.",
+            timeranges, required_vars)
```

Comment:
One question on the code here: Don't we want to raise an exception here? Does it make sense to continue with the code execution if this happens?